### PR TITLE
Rename spa.Module to spa.Network.

### PR DESCRIPTION
**Pull request by jgosmann**
**Motivation and context:**

Addresses nengo/nengo#1133.

There is no distinction between `spa.SPA` and `spa.Module` anymore. Furthermore, `spa.Module` can be used like any other `nengo.Network`. Thus, this renames `spa.Module` to `spa.Network`. If you intend to use SPA functionality, you can just switch to use `spa.Network` instead of `nengo.Network` everywhere.

**Interactions with other PRs:**

none

**How has this been tested?**

unit tests still pass

**How long should this take to review?**

- Average (neither quick nor lengthy)

**Where should a reviewer start?**

It's just the renaming and adjusting doc strings accordingly. Hopefully, quick to review even though many lines have been touched.

**Checklist:**

- [x] I have read the **CONTRIBUTING.rst** document.
- [x] I have updated the documentation accordingly.
- [n/a] I have included a changelog entry.
- [n/a] I have added tests to cover my changes.
- [x] All new and existing tests passed.

**Still to do:**

- [x] Check that no lines got longer than 80 characters.

### CODE REVIEW

**Comment by jgosmann** [comment]
Ready for review, hopefully a quick one as it is mostly just replacing `Module` with `Network`.

File: nengo_spa/input.py [hunk 1]
```diff
@@ -40,33 +40,33 @@ def input1(t):
...
        a vector of all zeros, and
-    2. an input to the ``task`` module which is always the value associated
+    2. an input to the ``task`` network which is always the value associated
        with the ``'X'`` semantic pointer.
 
     Parameters
     ----------
-    module : spa.Module, optional (Default: the current module)
+    network : spa.Module, optional (Default: the current SPA network)
         Module that this network provides input for.
```
Comment (tcstewar) [inline nengo_spa/input.py]: These two lines should have `Module` switched to `Network`

**Comment by tcstewar** [comment]
Other than the 2 missed Module->Network change sin the docstring in input.py (noted above), this looks great to me.  I notice that `spa_ast` still uses `Module` as a name, but I think it's fine to keep that.  Looks good to me!

**Comment by jgosmann** [comment]
> I notice that spa_ast still uses Module as a name, but I think it's fine to keep that.

Yeah, that's on purpose. Not all `spa_ast.Module`s are networks and having three different `Network`s might start to get confusing.

**Pull request merged**